deploy/terraform/modules/anydb_node/vm-anydb.tf

```
@@ -0,0 +1,139 @@
+  name                = format("%s_xdb_%02d-nic", local.anydb_sid, (count.index))
+  location            = var.resource-group[0].location
+  resource_group_name = var.resource-group[0].name
+
+  ip_configuration {
+    primary                       = true
+    name                          = "${local.dbnodes[count.index].name}-db-nic-ip"
+    subnet_id                     = local.sub_db_exists ? data.azurerm_subnet.anydb[0].id : azurerm_subnet.anydb[0].id
+    private_ip_address            = local.sub_db_exists ? local.dbnodes[count.index].db_nic_ip : lookup(local.dbnodes[count.index], "db_nic_ip", false) != false ? local.dbnodes[count.index].db_nic_ip : cidrhost(local.sub_db_prefix, tonumber(count.index) + 10)
+    private_ip_address_allocation = "static"
+  }
+}
+
+# Section for Linux Virtual machine 
+resource azurerm_linux_virtual_machine "dbserver" {
+  count                        = local.enable_deployment ? ((upper(local.anydb_ostype) == "LINUX") ? length(local.dbnodes) : 0) : 0
+  name                         = local.dbnodes[count.index].name
+  location                     = var.resource-group[0].location
+  resource_group_name          = var.resource-group[0].name
+  availability_set_id          = azurerm_availability_set.anydb[0].id
+  proximity_placement_group_id = local.ppgId
+  network_interface_ids        = [azurerm_network_interface.anydb[count.index].id]
+  size                         = local.dbnodes[count.index].size
+
+  source_image_id = local.anydb_custom_image ? local.anydb_os.source_image_id : null
+
+  dynamic "source_image_reference" {
+    for_each = range(local.anydb_custom_image ? 0 : 1)
+    content {
+      publisher = local.anydb_os.publisher
+      offer     = local.anydb_os.offer
+      sku       = local.anydb_os.sku
+      version   = local.anydb_os.version
+    }
+  }
+
+  dynamic "os_disk" {
+    iterator = disk
+    for_each = flatten([for storage_type in lookup(local.sizes, local.anydb_size).storage : [for disk_count in range(storage_type.count) : { name = storage_type.name, id = disk_count, disk_type = storage_type.disk_type, size_gb = storage_type.size_gb, caching = storage_type.caching }] if storage_type.name == "os"])
+    content {
+      name                 = format("%s-OsDisk", local.dbnodes[count.index].name)
+      caching              = disk.value.caching
+      storage_account_type = disk.value.disk_type
+      disk_size_gb         = disk.value.size_gb
+    }
+  }
+
+  computer_name                   = "${local.anydb_sid}dbl${count.index}"
+  admin_username                  = local.authentication.username
+  disable_password_authentication = local.authentication.type != "password" ? true : false
+
+  admin_ssh_key {
+    username   = local.authentication.username
+    public_key = file(var.sshkey.path_to_public_key)
+  }
+
+  boot_diagnostics {
+    storage_account_uri = var.storage-bootdiag.primary_blob_endpoint
+  }
+  tags = {
+    environment = "SAP"
+    SID         = local.anydb_sid
```

To be consistent with windows vm block.
```suggestion
    role        = "db"
    SID         = upper(local.anydb_sid)
```

---

fixed